BuildScript/BuildTasks/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,201 @@
+
+    [Required]
+    public ITaskItem[] SupportedTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledConfigurationIDs { get; set; }
+
+    [Output]
+    public ITaskItem[] Output { get; set; }
+
+    public override bool Execute ()
+    {
+      var supportedBrowsers = GetMetadataValueCollection (SupportedBrowsers, MetadataValueConstants.NoBrowser);
+      var supportedDatabaseSystems = GetMetadataValueCollection (SupportedDatabaseSystems, MetadataValueConstants.NoDB);
+      var supportedPlatforms = GetMetadataValueCollection (SupportedPlatforms);
+      var supportedExecutionRuntimes = GetMetadataValueDictionary (
+          SupportedExecutionRuntimes,
+          MetadataValueConstants.EnforcedLocalMachine,
+          MetadataValueConstants.LocalMachine);
+      var supportedTargetRuntimes = GetMetadataValueCollection (SupportedTargetRuntimes);
+
+      var testConfigurationsFactory = new TestConfigurationsFactory (
+          supportedBrowsers: supportedBrowsers,
+          supportedDatabaseSystems: supportedDatabaseSystems,
+          supportedPlatforms: supportedPlatforms,
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          supportedExecutionRuntimes: supportedExecutionRuntimes,
+          supportedTargetRuntimes: supportedTargetRuntimes);
+
+      var enabledBrowsers = GetMetadataValueCollection (EnabledBrowsers);
+      var enabledDatabaseSystems = GetMetadataValueCollection (EnabledDatabaseSystems);
+      var enabledPlatforms = GetMetadataValueCollection (EnabledPlatforms);
+      var enabledExecutionRuntimes = GetMetadataValueCollection (EnabledExecutionRuntimes);
+      var enabledTargetRuntimes = GetMetadataValueCollection (EnabledTargetRuntimes);
+      var enabledConfigurationIDs = GetMetadataValueCollection (EnabledConfigurationIDs);
+
+      var testConfigurationsFilter = new TestConfigurationsFilter (
+          supportedBrowsers: supportedBrowsers,
+          enabledBrowsers: enabledBrowsers,
+          supportedDatabaseSystems: supportedDatabaseSystems,
+          enabledDatabaseSystems: enabledDatabaseSystems,
+          supportedPlatforms: supportedPlatforms,
+          enabledPlatforms: enabledPlatforms,
+          supportedExecutionRuntimes: supportedExecutionRuntimes,
+          enabledExecutionRuntimes: enabledExecutionRuntimes,
+          supportedTargetRuntimes: supportedTargetRuntimes,
+          enabledTargetRuntimes: enabledTargetRuntimes,
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          enabledConfigurationIDs: enabledConfigurationIDs);
+
+      try
+      {
+        Output = Input.SelectMany (inputItem => CreateOutputTestConfigurations (inputItem, testConfigurationsFactory, testConfigurationsFilter)).ToArray();
+      }
+      catch (Exception ex) when (ex is InvalidOperationException || ex is ArgumentException)
```

Comment:
that change with the split catch I wanted to keep. see previous comment.

Reply:
Yeah, I just saw, I mixed things up, soz